### PR TITLE
Added separate python classes for blobs and chunks

### PR DESCRIPTION
**Pull request by MaciekPytel**
A whole bunch of related refactors in python API.
Updated examples/parse_bmp.py accordingly.

### CODE REVIEW

File: python/veles/objects/blob.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+from veles import network_pb2
+from veles.objects.base import LocalObject
+
+class Blob(LocalObject):
+    def __init__(self, client, proto_obj=None, id_path=None, parent=None):
+        super(Blob, self).__init__(client, proto_obj, id_path, parent)
+        self._client = client
```
Comment (MaciekPytel) [inline python/veles/objects/blob.py]: this should be in base class

File: python/examples/parse_bmp.py [hunk 2]
```diff
@@ -20,149 +20,119 @@ def parse_files(self):
...
-                '<i', file.data[self.client.cursor_value(file):
-                                self.client.cursor_value(file)+4])[0])
-            self.client.create_chunk(dib_header, 'Height', size=4)
-            self.client.create_chunk(dib_header, 'NumPlanes', size=2)
-            bits_per_pix = struct.unpack(
-                '<H', file.data[self.client.cursor_value(file):
-                                self.client.cursor_value(file)+2])[0]
-            self.client.create_chunk(dib_header, 'BitsPerPixel', size=2)
+            width_chunk = dib_header.create_chunk('Width', size=4)
+            width = struct.unpack('<i', width_chunk[0:4])[0]
```
Comment (altran01) [inline python/examples/parse_bmp.py]: Missing abs.

File: python/examples/parse_bmp.py [hunk 3]
```diff
@@ -20,149 +20,119 @@ def parse_files(self):
...
+            height_chunk = dib_header.create_chunk('Height', size=4)
+            height = struct.unpack('<i', height_chunk[0:4])[0]
```
Comment (altran01) [inline python/examples/parse_bmp.py]: Here as well.

Comment (MaciekPytel) [inline python/examples/parse_bmp.py]: oops... Will fix.

File: python/examples/parse_bmp.py [hunk 4]
```diff
@@ -20,149 +20,119 @@ def parse_files(self):
...
-                self.client.create_chunk(dib_header, 'XResolution', size=4)
-                self.client.create_chunk(dib_header, 'YResolution', size=4)
-                colors_used = struct.unpack(
-                    '<I',
-                    file.data[self.client.cursor_value(file):
-                              self.client.cursor_value(file)+4])[0]
-                self.client.create_chunk(dib_header, 'ColorsUsed', size=4)
-                self.client.create_chunk(dib_header, 'ColorsImportant', size=4)
+                compr_chunk = dib_header.create_chunk('Compression', size=4)
+                compression = struct.unpack('<i', compr_chunk[0:4])[0]
```
Comment (altran01) [inline python/examples/parse_bmp.py]: Compression is unsigned.

File: python/veles/objects/base.py [hunk 5]
```diff
@@ -100,3 +104,18 @@ def __repr__(self):
...
+        if start is None:
+            start = self._cursor
+
+        if end is None and size is None:
+            raise ValueError(
+                'You need to specify either size or end of chunk')
+        if end is None:
+            end = start + size
+        if update_cursor:
+            self._cursor = end
```
Comment (altran01) [inline python/veles/objects/base.py]: Hmm now that I think that it is better if we update cursor after we create chunk - that way if something fails and we want to retry from IPython we still have the old cursor value.

Comment (MaciekPytel) [inline python/veles/objects/base.py]: Good point, will do.

File: python/veles/objects/chunk.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+    def __iter__(self):
+        return self._blob[self._start:self._end].__iter__()
+
+    def __getitem__(self, arg):
+        if self._default_address_mode == self.AddressMode.ABSOLUTE:
+            return self._blob[arg]
+        if isinstance(arg, int):
+            return self._blob(arg + self._start)
+        if isinstance(arg, slice):
```
Comment (altran01) [inline python/veles/objects/chunk.py]: What do you think about adding possibility to use `chunk[:]` to get whole chunk content ? I think it is more convenient than `chunk[0:4]`

File: python/veles/veles_api.py [hunk 7]
```diff
@@ -29,11 +30,18 @@ class ObjectTypes(object):
         SUB_BLOB = 2
         CHUNK = 3
 
+        CONSTRUCTORS = [
+            objects.LocalObject,  # ROOT
+            objects.Blob,  # FILE_BLOB
+            objects.Blob,  # SUB_BLOB
+            objects.Chunk  # CHUNK
```
Comment (altran01) [inline python/veles/veles_api.py]: I think dictionary type->class would be more appropriate, what do you think?

File: python/veles/objects/chunk.py (hunk 6 above)
Comment (MaciekPytel) [inline python/veles/objects/chunk.py]: I don't see why not - will fix.

File: python/veles/veles_api.py (hunk 7 above)
Comment (MaciekPytel) [inline python/veles/veles_api.py]: Fair enough, might as well not duplicate enums.

**Pull request merged**